### PR TITLE
Strict content types in request/response

### PR DESCRIPTION
**Pull request by alaski**
Strict validation is added to the server boot POST handler so that only
requests with content_type application/json are allowed.  Additionally
the error handlers are updated to return application/json responses
rather than the default of text/html.

### CODE REVIEW

File: enamel/main.py [hunk 1]
```diff
@@ -66,10 +79,13 @@ def home():
 
 
 def server_boot():
+    if flask.request.content_type != 'application/json':
+        flask.abort(400)
```
Comment (cdent) [inline enamel/main.py]: You want 415 here I think: https://httpstatuses.com/415

Comment (cdent) [inline enamel/main.py]: Also, not for now, but eventually this is probably something we're going to want for most methods so we may want to find a decorator or something that enforces this. I would guess that something is already in the flask stack the does it.

Comment (alaski) [inline enamel/main.py]: Thanks.  I agree that this is the wrong place to enforce it.  I'll add a comment noting that.

**Comment by jaypipes** [comment]
@alaski It looks like @cdent beat you to this, apologies :(

See his recent commit here: https://github.com/jaypipes/enamel/pull/25

**Comment by cdent** [comment]
Yeah, sorry about that. I went with what I did because I felt like it was more expressive, despite being somewhat non-flasky. I want us to be explicit that we know which HTTP status code we are choosing to raise when we do.

**Comment by cdent** [comment]
I reckon a decorator is probably a good idea.

**Comment by alaski** [comment]
No worries.

**Pull request closed without merging**